samples/EverythingServer/Program.cs

```diff
@@ -0,0 +1,195 @@
+            ]
+        });
+    })
+    .WithReadResourceHandler((ctx, ct) =>
+    {
+        var uri = ctx.Params?.Uri;
+
+        if (uri is null || !uri.StartsWith("test://static/resource/"))
+        {
+            throw new NotSupportedException($"Unknown resource: {uri}");
+        }
+
+        int index = int.Parse(uri["test://static/resource/".Length..]) - 1;
+
+        if (index < 0 || index >= ResourceGenerator.Resources.Count)
+        {
+            throw new NotSupportedException($"Unknown resource: {uri}");
+        }
+
+        var resource = ResourceGenerator.Resources[index];
+
+        if (resource.MimeType == "text/plain")
+        {
+            return Task.FromResult(new ReadResourceResult
+            {
+                Contents = [new TextResourceContents
+                {
+                    Text = resource.Description!,
+                    MimeType = resource.MimeType,
+                    Uri = resource.Uri,
+                }]
+            });
+        }
+        else
+        {
+            return Task.FromResult(new ReadResourceResult
+            {
+                Contents = [new BlobResourceContents
+                {
+                    Blob = resource.Description!,
+                    MimeType = resource.MimeType,
+                    Uri = resource.Uri,
+                }]
+            });
+        }
+    })
+    .WithSubscribeToResourcesHandler(async (ctx, ct) =>
+    {
+        var uri = ctx.Params?.Uri;
+
+        if (uri is not null)
+        {
+            subscriptions.Add(uri);
+
+            await ctx.Server.RequestSamplingAsync([
+                new ChatMessage(ChatRole.System, "You are a helpful test server"),
+                new ChatMessage(ChatRole.User, $"Resource {uri}, context: A new subscription was started"),
+            ],
+            options: new ChatOptions
+            {
+                MaxOutputTokens = 100,
+                Temperature = 0.7f,
+            },
+            cancellationToken: ct);
+        }
+
+        return new EmptyResult();
+    })
+    .WithUnsubscribeFromResourcesHandler((ctx, ct) =>
+    {
+        var uri = ctx.Params?.Uri;
+        if (uri is not null)
+        {
+            subscriptions.Remove(uri);
+        }
+        return Task.FromResult(new EmptyResult());
+    })
+    .WithGetCompletionHandler((ctx, ct) =>
+    {
+        var exampleCompletions = new Dictionary<string, IEnumerable<string>>
+        {
+            { "style", ["casual", "formal", "technical", "friendly"] },
+            { "temperature", ["0", "0.5", "0.7", "1.0"] },
+            { "resourceId", ["1", "2", "3", "4", "5"] }
+        };
+
+        var @ref = ctx.Params?.Ref;
+
+        if (@ref is null)
+        {
+            throw new NotSupportedException($"Reference is required.");
+        }
+
+        var argument = ctx.Params!.Argument;
```

Comment:
Nit:

```suggestion
        if (ctx.Params is not { } params)
        {
            throw new NotSupportedException($"Params are required.");
        }
        
        var @ref = params.Ref;
        var argument = params.Argument;
```

@eiriktsarpalis should we make RequestContext.Params non nullable? `TParams` could always be made nullable if that's what we really want.